### DISCUSSION TITLE
Why do we adopt array alignment?

### DISCUSSION THREAD

**Discussion by mohamedsabil83** [Laravel-Lang/common#53] [Q&A] [answered]
I was asking myself this question a while ago? Why do we adopt array alignment even though Laravel does not?

**Comment by luisprmat** [comment]
Are you referring to this?

![image](https://user-images.githubusercontent.com/9275870/101280245-0b59f900-3796-11eb-9d3e-9acc295e2b33.png)

**Reply by mohamedsabil83** [reply]
Yes, I do. It's matching how Laravel did. Also, if we adding a new long phrase in the future, we don't need to re-align all the lines.

**Reply by luisprmat** [reply]
I agree with you, it's better like in Laravel. Imagine that we also align the JSON, it would look awful!!!
```json
{
    ...
    "This user already belongs to the team.": "Este usuario ya pertenece al equipo.",
    "to":                                     "al",
    "Toggle navigation":                      "Activar navegación",
    "Token Name":                             "Nombre del token",
    ...
}
```
I vote to change the alignment we have and use Laravel's, and also to be consistent.
@caouecs, @masterix21, @WhereIsLucas, @jorgercosta, @andrey-helldar, @LukaLatkovic, @MrAlKuz, @overtrue, @jensstigaard,  and team, what do you think about this?

**Reply by andrey-helldar** [reply]
> Yes, I do. It's matching how Laravel did. Also, if we adding a new long phrase in the future, we don't need to re-align all the lines.

Formatting lines is done by pressing three buttons on the keyboard - Ctrl+Alt+L ([phpStorm](https://www.jetbrains.com/help/phpstorm/reformat-and-rearrange-code.html))

**Reply by andrey-helldar** [reply]
> I agree with you, it's better like in Laravel. Imagine that we also align the JSON, it would look awful!!!

Therefore, we do not touch JSON.

> I vote to change the alignment we have and use Laravel's, and also to be consistent.

I am for beautiful code. Without alignment, the code is hard to read. It seems that such code is written by those who did not study well at school.

See:
![image](https://user-images.githubusercontent.com/10347617/101287466-7c82c600-3801-11eb-913d-fc9aa66b956f.png)

versus
![image](https://user-images.githubusercontent.com/10347617/101287476-89071e80-3801-11eb-86c2-6622b8d2288a.png)

As I've said many times, what Taylor does is good, but far from ideal.

**Reply by andrey-helldar** [reply]
@luisprmat, @mohamedsabil83, there is another way to solve problem - to use a translation manager. For example, [this one](https://github.com/andrey-helldar/laravel-lang-publisher). (@caouecs, for example 😀 ).

In his [settings](https://github.com/andrey-helldar/laravel-lang-publisher/blob/master/config/lang-publisher.php#L21-L27), you can set the [alignment option](https://github.com/andrey-helldar/laravel-lang-publisher/blob/master/config/lang-publisher.php#L21-L27). If it is not needed, set `'alignment' => false` and the files will be saved in Laravel-way.

**Comment by andrey-helldar** [comment, marked as answer]
Because Taylor doesn't care what the code looks like - good or bad.

We prefer pretty code and therefore we apply alignment in php files.

**Reply by mohamedsabil83** [reply]
> Formatting lines is done by pressing three buttons on the keyboard - Ctrl+Alt+L (phpStorm)

Yes, that's what I do.

> Because Taylor doesn't care what the code looks like - good or bad.
> 
> We prefer pretty code and therefore we apply alignment in php files.

As I guess. Just to know if there is another reason. I'll keep the package aligned and ignore alignment on my projects

**Reply by andrey-helldar** [reply]
No, there is no other reason. Only the visual beauty of the code. 😊

**Comment by masterix21** [comment]
I agree with @andrey-helldar because we must do everything to simplify our work, keep a clean code, and maintainable.
